osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -184,8 +184,11 @@ protected override void LoadAsyncComplete()
         {
             base.LoadAsyncComplete();
 
-            if (HitObject != null)
-                Apply(HitObject, lifetimeEntry);
+            if (initialHitObject != null)
+            {
+                Apply(initialHitObject, null);
+                initialHitObject = null;
```

Comment:
Don't really like this `initialHitObject` song-and-dance. Wouldn't it work better to just create the unmanaged lifetime entry in the ctor and just apply it if it's not null here?

Reply:
I had to introduce a state of `lifetimeEntry != null && !hasEntryApplied` so it is not "DHO is applied iff `lifetimeEntry != null`" anymore but I think it is fine.